Makefile

```diff
@@ -17,12 +17,21 @@ k0sctl: $(GO_SRCS)
 bin/k0sctl-linux-amd64: $(GO_SRCS)
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build $(BUILD_FLAGS) -o bin/k0sctl-linux-amd64 main.go
 
+docker/linux-amd64: bin/k0sctl-linux-amd64
```

Comment:
Something like this could work

```Makefile
.PHONY: docker/build
docker/build:
	docker buildx build \
	  --build-arg ENVIRONMENT=$(ENVIRONMENT) \
	  --build-arg GIT_COMMIT=$(GIT_COMMIT) \
	  --build-arg TAG_NAME=$(TAG_NAME) \
	  -t ghcr.io/k0sproject/k0sctl:$(TAG_NAME) \
		--load \
		.

.PHONY: docker/push
docker/push: 
	docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v7 \
	  --build-arg ENVIRONMENT=$(ENVIRONMENT) \
	  --build-arg GIT_COMMIT=$(GIT_COMMIT) \
	  --build-arg TAG_NAME=$(TAG_NAME) \
	  -t ghcr.io/k0sproject/k0sctl:$(TAG_NAME) \
		--push \
		.
```